compiler/rustc_lint/src/non_fmt_panic.rs

```diff
@@ -69,23 +69,62 @@ fn check_panic<'tcx>(cx: &LateContext<'tcx>, f: &'tcx hir::Expr<'tcx>, arg: &'tc
 
     let (span, panic) = panic_call(cx, f);
 
-    cx.struct_span_lint(NON_FMT_PANIC, arg.span, |lint| {
+    // Find the span of the argument to `panic!()`, before expansion in the
+    // case of `panic!(some_macro!())`.
```

Comment:
I had to think to realize why you didn't use `source_callsite`.
```suggestion
    // Find the span of the argument to `panic!()`, before expansion in the
    // case of `panic!(some_macro!())`.
    // NOTE: the panic call may itself be inside another macro.
```